src/indicators/idmc/run.R

```diff
@@ -38,303 +54,124 @@ df_idmc_raw <- idmc_get_data() %>%
+    date,
+    displacement_daily
   ) %>%
   group_by(
-    iso3,
-    flag_name,
-    cs_temp_
-  ) %>%
-  summarize(
-    start_date = min(date),
-    end_date = max(date),
-    .groups = "drop_last"
-  ) %>%
-  select(
-    -cs_temp_
-  ) %>%
-  mutate(
-    uuid = row_number()
-  ) %>%
-  ungroup() %>%
-  left_join(
-    df_links,
-    by = "iso3"
-  ) %>%
-  mutate(
-    url = paste0("https://www.internal-displacement.org/countries/", country_link)
-  ) %>%
-  select(
-    -country_link
-  )
-
-# get total displacement within each of the flags
-# difficult for any flag because start date depends on what kind of flags have
-# generated the flags within it
-
-df_flags_start <- df_idmc_flags %>%
-  filter(
-    flag_name != "flag_idmc_any"
-  ) %>%
-  mutate(
-    flag_start_date = start_date,
-    start_date = case_when(
-      str_detect(flag_name, "yearly") ~ start_date - days(364),
-      str_detect(flag_name, "quarterly") ~ start_date - days(89),
-      str_detect(flag_name, "monthly") ~ start_date - days(29),
-      str_detect(flag_name, "weekly") ~ start_date - days(6),
-      TRUE ~ start_date
-    )
+    iso3
   )
 
-# setting flagging priority for the latest flag for a country
-# we start with the 1st in a period as priority
-# then look at weekly -> yearly in increasing priority so that user sees
-# the more acute, short term fluctuations on the main dashboard page
-flag_priority <- c("flag_idmc_1st_year", "flag_idmc_1st_6_months",
-                   "flag_idmc_1st_3_months", "flag_idmc_global_weekly",
-                   "flag_idmc_weekly", "flag_idmc_global_monthly",
-                   "flag_idmc_monthly", "flag_idmc_global_quarterly",
-                   "flag_idmc_quarterly", "flag_idmc_global_yearly",
-                   "flag_idmc_yearly", "flag_idmc_any")
+# calculate the flags using the utilities functions
+df_flagged <- calculate_flags(
+  .data = df_idmc,
+  date = 1,
+  x = displacement_daily,
+  first_since = c(180, 365),
+  periods = c(7, 30, 90, 365),
+  thresholds_pcts = 0.95,
+  thresholds_static = c(5000, 25000, 100000, 500000),
+  thresholds_minimums = c(1000, 5000, 10000, 25000)
+)
 
-# for any flags, join up to the other data set
-# and make the start date the minimum of the start date of any
-# other flag in that time frame
-df_flags_start_any <- df_idmc_flags %>%
-  filter(
-    flag_name == "flag_idmc_any"
-  ) %>%
-  full_join(
-      select(
-        df_flags_start,
-        iso3,
-        flag_name2 = flag_name,
-        start_date2 = start_date,
-        end_date2 = end_date,
-        flag_start_date
-      ),
-    by = "iso3",
-    multiple = "all",
-    relationship = "many-to-many"
-  ) %>%
-  group_by(iso3, flag_name, uuid) %>%
-  filter(
-    end_date2 >= start_date,
-    flag_start_date <= end_date
-  ) %>%
-  mutate(
-    start_date = min(start_date2)
-  ) %>%
-  filter(
-    end_date2 == end_date
+df_idmc_flags <- wrangle_alerts(
+  df_flagged,
+  date,
+  displacement_daily
```

Comment:
- I think `data-masking` in this context may add unnecessary complexity making debugging and maintenance more difficult in the long-run (see comment around line 267).
- `df_idmc_flags` - returns some NA values for `latest_flag` - is this right?

Reply:
>  every time I use it I just regret it, and it makes things unnecessarily complex.

LOL - I was literally going to write the same exact thing!